variables.tf

```diff
@@ -1,4 +1,118 @@
-variable "example" {
-  description = "Example variable"
-  default     = "hello world"
+variable "zone" {
+  type        = string
+  description = "The DNS zone name which will be added."
+}
+
+variable "zone_enabled" {
+  type        = bool
+  description = "Whether to create DNS zone otherwise use existing."
+  default     = true
+}
+
+variable "records" {
+  type        = list(any)
+  default     = null
+  description = <<-DOC
+    name:
+      The name of the record.
+    type:
+      The type of the record.
+    value:
+      The value of the record.
+    ttl:
+      The TTL of the record.
+      Default value: 1.
+    priority:
+      The priority of the record. 
+    proxied:
+      Whether the record gets Cloudflare's origin protection. 
+      Default value: false.
+  DOC
+}
+
+variable "paused" {
+  type        = bool
+  description = "Whether this zone is paused (traffic bypasses Cloudflare)"
+  default     = false
+}
+
+variable "jump_start" {
+  type        = bool
+  description = "Whether to scan for DNS records on creation."
+  default     = false
+}
+
+variable "plan" {
+  type        = string
+  description = "The name of the commercial plan to apply to the zone. Possible values: `free`, `pro`, `business`, `enterprise`"
+  default     = "free"
+}
+
+variable "type" {
+  type        = string
+  description = "A full zone implies that DNS is hosted with Cloudflare. A `partial` zone is typically a partner-hosted zone or a CNAME setup. Possible values: `full`, `partial`."
+  default     = "full"
+}
+
+variable "argo_enabled" {
+  type        = bool
+  description = "Whether to enable Cloudflare Argo for DNS zone"
+  default     = false
+}
+
+variable "argo_tiered_caching_enabled" {
+  type        = bool
+  description = "Whether tiered caching is enabled."
+  default     = true
+}
+
+variable "argo_smart_routing_enabled" {
+  type        = bool
+  description = "Whether smart routing is enabled."
+  default     = true
+}
+
+variable "healthchecks" {
+  type        = list(any)
+  default     = null
+  description = <<-DOC
+  A list of maps of Health Checks rules. 
+  The values of map is fully compliant with `cloudflare_healthcheck` resource.
+  To get more info see https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/healthcheck
+  DOC
+}
+
+variable "firewall_rules" {
+  type        = list(any)
```

Comment:
same as above...consider `set(object(...))` or `list(object(...))`

Reply:
I agree about using `any` instead of `object` prior to TF 0.15, but I think we should `map(any)` and use the keys for names.